### ISSUE TITLE
Memory leak/inconsistency in DragGestureRecognizer/OneSequenceGestureRecognizer

### ISSUE THREAD

**Issue by spkersten** [flutter/flutter#84040]
I discovered this while using `DragGestureRecognizer` as a starting point for a new recogniser.

In:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/gestures/recognizer.dart#L246
Add for diagnostics:
```dart
  @override
  bool isPointerAllowed(PointerDownEvent event) {
    print('_entries: ${_entries.length}');
    return super.isPointerAllowed(event);
  }
```

With the following minimal example:

```dart
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => Container(
        color: Color(0xffff0000),
        child: Center(
          child: GestureDetector(
            onPanDown: (_) => print('onPanDown'),
            onPanStart: (_) => print('onPanStart'),
            onPanUpdate: (_) => print('onPanUpdate'),
            onPanEnd: (_) => print('onPanEnd'),
            onPanCancel: () => print('onPanCancel'),
            child: Container(
              width: 300,
              height: 300,
              color: Color(0xff00ff00),
            ),
          ),
        ),
      );
}
```

Single tap the container, a few times and the output is:
```
flutter: _entries: 0
flutter: onPanDown
flutter: onPanStart
flutter: onPanUpdate
flutter: onPanEnd
flutter: _entries: 1
flutter: onPanDown
flutter: onPanStart
flutter: onPanUpdate
flutter: onPanEnd
flutter: _entries: 2
flutter: onPanDown
flutter: onPanStart
flutter: onPanUpdate
flutter: onPanEnd
flutter: _entries: 3
flutter: onPanDown
flutter: onPanStart
flutter: onPanUpdate
flutter: onPanEnd
```

```
[✓] Flutter (Channel stable, 2.2.0, on macOS 11.3.1 20E241 darwin-x64, locale en-NL)
    • Flutter version 2.2.0 at /Users/spkersten/Development/flutter
    • Framework revision b22742018b (3 weeks ago), 2021-05-14 19:12:57 -0700
    • Engine revision a9d88a4d18
    • Dart version 2.13.0
```

**Comment by TahaTesser** [comment]
Hi @spkersten 
Thanks for filing the issue

I can see the same results as you on the stable, however on master, i see the result below

```
[   +2 ms] Syncing files to device macOS... (completed in 105ms)
[   +1 ms] Synced 0.0MB.
[   +1 ms] <- accept
[   +6 ms] Connected to _flutterView/0x7f91cc855a20.
[   +3 ms] Flutter run key commands.
[   +2 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[   +1 ms] An Observatory debugger and profiler on macOS is available at: http://127.0.0.1:60992/wWN143015js=/
[ +568 ms] DevTools activation throttled until 2021-06-07 21:29:49.613602.
[ +518 ms] The Flutter DevTools debugger and profiler on macOS is available at:
                    http://127.0.0.1:9104?uri=http%3A%2F%2F127.0.0.1%3A60992%2FwWN143015js%3D%2F
[+208584 ms] flutter: _entries: 0
[  +33 ms] flutter: onPanDown
[   +4 ms] flutter: onPanStart
[   +8 ms] flutter: onPanEnd
[ +232 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +87 ms] flutter: onPanEnd
[  +89 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +78 ms] flutter: onPanEnd
[  +77 ms] flutter: _entries: 0
[   +2 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +78 ms] flutter: onPanEnd
[  +76 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +64 ms] flutter: onPanEnd
[ +103 ms] flutter: _entries: 0
[   +2 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +65 ms] flutter: onPanEnd
[  +91 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +65 ms] flutter: onPanEnd
[  +83 ms] flutter: _entries: 0
[   +2 ms] flutter: onPanDown
[        ] flutter: onPanStart
[ +112 ms] flutter: onPanEnd
[  +78 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[   +1 ms] flutter: onPanEnd
[  +90 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +53 ms] flutter: onPanEnd
[+2530 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +52 ms] flutter: onPanUpdate
[  +11 ms] flutter: onPanEnd
[  +53 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +79 ms] flutter: onPanUpdate
[        ] flutter: onPanEnd
[  +68 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +78 ms] flutter: onPanUpdate
[        ] flutter: onPanEnd
[  +65 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +66 ms] flutter: onPanUpdate
[        ] flutter: onPanEnd
[  +66 ms] flutter: _entries: 0
[        ] flutter: onPanDown
[        ] flutter: onPanStart
[  +67 ms] flutter: onPanUpdate
[        ] flutter: onPanEnd
[  +71 ms] flutter: _entries: 0
[   +2 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +71 ms] flutter: onPanUpdate
[        ] flutter: onPanEnd
[  +48 ms] flutter: _entries: 0
[   +1 ms] flutter: onPanDown
[        ] flutter: onPanStart
[  +53 ms] flutter: onPanUpdate
[        ] flutter: onPanEnd
```

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel master, 2.3.0-17.0.pre.226, on macOS 11.4 20F71 darwin-x64,
    locale en-GB)
    • Flutter version 2.3.0-17.0.pre.226 at
      /Users/tahatesser/Code/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c9d5485361 (13 hours ago), 2021-06-06 18:59:02 -0400
    • Engine revision 66068e2e7d
    • Dart version 2.14.0 (build 2.14.0-186.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Volumes/Extreme/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile)     • e534bcc711c9a011aa892eb51cdb60f40b3acddb •
      ios            • iOS 12.5.2
    • iPhone 12 Pro Max (mobile) • 516FAD15-C23C-4952-943C-0612DAAC1643     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • macOS 11.4 20F71 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 91.0.4472.77

• No issues found!
```

</details>

Can you please upgrade to the latest  `master` and try again
Execute the following commands to upgrade
`flutter channel master`
`flutter upgrade --force`
`flutter doctor -v`

**Comment by spkersten** [comment]
Cool. Seems to be fixed then.
